sockeye/optimizers.py

```diff
@@ -35,8 +35,8 @@ class SockeyeOptimizer(Optimizer):
     when updating weights.
     """
     def __init__(self, **kwargs) -> None:
-        self.batch_state: BatchState = None
-        self.checkpoint_state: CheckpointState = None
+        self.batch_state = None
```

Comment:
You can leave the type annotations but change them to Optional[BatchState] to allow None (then typechecking will also work :))

Reply:
Good to know!